components/molecule/phoneValidation/src/index.js

```diff
@@ -0,0 +1,157 @@
+        <div
+          ref={inputPrefixRef}
+          className={`${baseClass}-input-prefix`}
+          testId="molecule-phone-validation-input-prefix"
+          onClick={() => setShowDropdown(!showDropdown)}
+        >
+          <span className={`${baseClass}-input-prefix-flag`}>
+            {getFlagEmoji(selectedPrefix.value)}
+          </span>
+          <AtomIcon
+            size={ATOM_ICON_SIZES.medium}
+            color={ATOM_ICON_COLORS.currentColor}
+          >
+            {dropdownIcon}
+          </AtomIcon>
+        </div>
+        <div className={`${baseClass}-input-phoneContainer`}>
+          <p className={`${baseClass}-input-prefix-code`}>
+            {selectedPrefix.countryCode}
+          </p>
+          <input
+            className={`${baseClass}-input-phone`}
+            type="tel"
+            maxLength={maxLength}
+            value={
+              phone.split(' ')[1]
+                ? phone
+                    .split(' ')[1]
+                    .replace(/\s/g, '')
+                    .match(/.{1,3}/g)
+                    .join(' ')
+                : ''
+            }
+            onChange={e =>
+              setPhone(
+                `${selectedPrefix.countryCode} ${e.target.value.replace(
+                  /\s/g,
+                  ''
+                )}`
+              )
+            }
+          />
+        </div>
+      </div>
+
+      <MoleculeDropdownList ref={modalRef} visible={showDropdown}>
+        {prefixes.map(prefix => {
+          return (
+            <MoleculeDropdownOption
+              value={prefix.value}
+              selected={selectedPrefix}
+              key={prefix.countryCode}
+              onClick={() => {
+                setSelectedPrefix(prefix)
+                setShowDropdown(false)
+              }}
+            >
+              <div className={`${baseClass}-dropdown-option`}>
+                <span className={`${baseClass}-dropdown-option-flag`}>
+                  {getFlagEmoji(prefix.value)}
+                </span>
+                <span
+                  className={
+                    prefix === selectedPrefix
+                      ? `${baseClass}-dropdown-option-label selected`
+                      : `${baseClass}-dropdown-option-label`
+                  }
+                >
+                  {prefix.label}
+                </span>
+                <span className={`${baseClass}-dropdown-option-code`}>
+                  ({prefix.countryCode})
+                </span>
+              </div>
+            </MoleculeDropdownOption>
+          )
+        })}
+      </MoleculeDropdownList>
+    </div>
+  )
+}
+
+MoleculePhoneValidation.displayName = 'MoleculePhoneValidation'
+
+MoleculePhoneValidation.propTypes = {
+  dropdownIcon: PropTypes.node,
+  maxLength: PropTypes.number,
+  phone: PropTypes.string,
+  prefixes: PropTypes.array,
+  setPhone: PropTypes.func,
+  type: PropTypes.oneOf(Object.values(phoneValidationType))
+}
```

Comment:
when I first saw the name of the component, `molecule/phoneValidation` I thought it was some sort of wrapper / helper component specific for validation of phone numbers... it is that, and it's also an `input` as I understand it. Would it make sense to you to name it something like `molecule/inputPhoneNumber` or `molecule/inputPhoneValidation`...?

Reply:
PhoneInput

Reply:
I thought about it too, if you both think that way I'm gonna rename it, thks for the feedback <3